src/main/java/reposense/parser/CsvParser.java

```diff
@@ -76,6 +77,23 @@ public CsvParser(Path csvFilePath) throws IOException {
         return results;
     }
 
+    /**
+     * Returns an array of Strings representing the header of a CSV file.
+     *
+     * @param reader A BufferedReader opened on the CSV file.
+     * @return an array of Strings where each Strings is a column name from the CSV header.
+     * @throws IOException
+     */
```

Comment:
JavaDocs are optional for `private` methods but good to have.

Generally, avoid trivial or empty `@annotation`. 
You don't have to write everything but try to capture the precise behavior of the method.

```
/**
 * Returns the first non-empty line as an array of Strings representing
 * the CSV header row.
 *
 * @throws IOException if there are error accessing the given csv file.
 */
```